pkg/resource/deploy/deploytest/resourcemonitor.go

```diff
@@ -273,9 +274,10 @@ func (rm *ResourceMonitor) RegisterResource(t tokens.Type, name string, custom b
 	// that are erroneously returned (e.g. secrets/resource refs that are returned even though the caller has not set
 	// `AcceptSecrets` or `AcceptResources` to `true` above).
 	outs, err := plugin.UnmarshalProperties(resp.Object, plugin.MarshalOptions{
-		KeepUnknowns:  true,
-		KeepSecrets:   true,
-		KeepResources: true,
+		KeepUnknowns:     true,
+		KeepSecrets:      true,
+		KeepResources:    true,
+		KeepOutputValues: true,
```

Comment:
Is this set for the same reason as `KeepSecrets` and `KeepResources` from the comment above? Otherwise, I'm not following why it's needed.